app/models/close_reason.rb

```diff
@@ -0,0 +1,6 @@
+# Represents a close reason. Close reasons can be assigned to posts
+
+class CloseReason < ApplicationRecord
+  belongs_to :community
+  scope :active, ->(community_id) { where(active: true, community_id: nil).or(where(active: true, community_id: community_id)) }
```

Comment:
On L4, use `include CommunityRelated` - that'll apply the `belongs_to :community` and also apply a default scope so that queries only get the current community's data by default.

That will also mean you need to change L5 to use `unscoped.where` instead of just `where`.